### ISSUE TITLE
Batch normalization with codegen

### ISSUE THREAD

**Issue by Nephilim02** [ZantFoundation/Z-Ant#189]
### Feature Description

Based on the documentation:
https://onnx.ai/onnx/operators/onnx__BatchNormalization.html

Implement in the appropriate file in /TensorMath:

- batch_normalization
- lean_batch_normalization
- get_batch_normalization_output_shape

Implement in codegen_math_handler:

- write_batch_normalization (add call in write_math_op)
- compute_batch_normalization_output_shape (add call in compute_output_shape)
